edx_sga/sga.py

```diff
@@ -0,0 +1,530 @@
+                    needs_approval = score is not None
+                else:
+                    needs_approval = False
+                instructor = self.is_instructor()
+                yield {
+                    'module_id': module.id,
+                    'student_id': student.student_id,
+                    'submission_id': submission['uuid'],
+                    'username': module.student.username,
+                    'fullname': module.student.profile.name,
+                    'filename': submission['answer']["filename"],
+                    'timestamp': submission['created_at'].strftime(
+                        DateTime.DATETIME_FORMAT
+                    ),
+                    'score': score,
+                    'approved': approved,
+                    'needs_approval': instructor and needs_approval,
+                    'may_grade': instructor or not approved,
+                    'annotated': state.get("annotated_filename"),
+                    'comment': state.get("comment", ''),
+                }
+
+        return {
+            'assignments': list(get_student_data()),
+            'max_score': self.max_score(),
+        }
+
+    def studio_view(self, context=None):
+        try:
+            cls = type(self)
+
+            def none_to_empty(x):
+                return x if x is not None else ''
+            edit_fields = (
+                (field, none_to_empty(getattr(self, field.name)), validator)
+                for field, validator in (
+                    (cls.display_name, 'string'),
+                    (cls.points, 'number'),
+                    (cls.weight, 'number'))
+            )
+
+            context = {
+                'fields': edit_fields
+            }
+            fragment = Fragment()
+            fragment.add_content(
+                render_template(
+                    'templates/staff_graded_assignment/edit.html',
+                    context
+                )
+            )
+            fragment.add_javascript(_resource("static/js/src/studio.js"))
+            fragment.initialize_js('StaffGradedAssignmentXBlock')
+            return fragment
+        except:  # pragma: NO COVER
+            log.error("Don't swallow my exceptions", exc_info=True)
+            raise
+
+    @XBlock.json_handler
+    def save_sga(self, data, suffix=''):
+        self.display_name = data.get('display_name', self.display_name)
+        self.weight = data.get('weight', self.weight)
+
+        # Validate points before saving
+        points = data.get('points', self.points)
+        # Check that we are an int
+        try:
+            points = int(points)
+        except ValueError:
+            raise JsonHandlerError(400, 'Points must be an integer')
+        # Check that we are positive
+        if points < 0:
+            raise JsonHandlerError(400, 'Points must be a positive integer')
+        self.points = points
+
+    @XBlock.handler
+    def upload_assignment(self, request, suffix=''):
```

Comment:
This approach will have all file uploads going through our servers.  That may have some performance implications that we should be careful about, and we may want to explicitly limit file sizes if we go this route. 

Another approach is to leverage the FileUpload API in ORA2. That approach is currently, unfortunately, limited to S3 uploads, but interactions with S3 directly rather than having file uploads go through our servers

If this is out of scope, that's fine, but I'd like to just make note of it, if we encounter large uploads bogging down our servers.